docs/java/features/resilience/resilience.md

```diff
@@ -0,0 +1,201 @@
+
+:::note
+Note that the Resilience Decorator will try to propagate the current [Thread Context](../multi-tenancy/thread-context.md) at the _time the decorator is invoked_. This is important when you are decorating a Callable or Supplier and running it later. The Thread Context must be available whenever `decorateCallable` or `decorateSupplier` is evaluated. So if the call to `ResilienceDecorator` should take place asynchronously you should [follow these steps](../multi-tenancy/thread-context.md#running-asynchronous-operations) to ensure the Thread Context is available.
+:::
+
+#### Failures and Fallbacks
+
+An operation might fail for two reasons:
+
+1. The operation itself encounters a failure and throws an error or exception
+2. A resilience pattern causes the operation to fail (e.g. the circuit breaker prevents further invocations)
+
+The SDK wraps all kind of checked and unchecked exceptions into a `ResilienceRuntimeException` and throws them.
+
+To deal with failures one can either catch the `ResilienceRuntimeException` or provide a fallback function:
+
+```java
+executeCallable(() -> operation(), configuration,
+(throwable) -> {
+    log.debug("Encountered a failure in operation: ", throwable);
+    log.debug("Proceeding with fallback value: {}", fallback);
+    return fallback;
+});
+```
+
+In the case of Callable this relieves you of the need to catch the exception at the outer level.
+
+### Building a Resilience Configuration
+
+A new `ResilienceConfiguration` _with default values_ is created by providing an identifier for it:
+
+```java
+configuration = ResilienceConfiguration.of("identifier");
+```
+
+The identifier can be either a string or a class.
+In case of the latter the (full) classname will be used as identifier.
+
+Check [the JavaDoc](https://help.sap.com/http.svc/rc/b579bf8578954412aea2b458e8452201/1.0/en-US/com/sap/cloud/sdk/cloudplatform/resilience/ResilienceConfiguration.html#of-java.lang.String-) for which patterns and parameters will be applied by default.
+You can also create a configuration with all patterns disabled:
+
+```java
+configuration = ResilienceConfiguration.empty("identifier");
+```
+
+Individual resilience patterns are configured via dedicated builder classes like `TimeLimiterConfiguration` and are added to the configuration via dedicated setters, e.g. `timeLimiterConfiguration()`.
+For details see the list of [Resilience Capabilities](#resilience-capabilities) below.
+
+#### Multi Tenancy
+
+The SDK is capable of applying the different resilience patterns in a tenant and principal aware manner.
+Consider for example the Bulkhead pattern which limits the amount of parallel executions.
+If the operation is tenant specific one would probably want to avoid one tenant blocking all others.
```

Comment:
```suggestion
If the operation is tenant specific then you would probably want to avoid one tenant blocking all others.
```

Reply:
Is there a specific reason why you would suggest `you` over `one`?

Reply:
In my first read, the sentence seemed to me a bit unclear because of "one" occurring twice. In general, I dont mind using one/you/we (either of these).

Reply:
Good point!